opentelemetry-sdk/benches/log_exporter.rs

```diff
@@ -6,8 +6,8 @@
     RAM: 64.0 GB
     | Test                           | Average time|
     |--------------------------------|-------------|
-    | LogExporterWithFuture          | 122 ns      |
-    | LogExporterWithoutFuture       | 89 ns      |
+    | LogExporterWithFuture          | 108 ns      |
+    | LogExporterWithoutFuture       | 88 ns      |
```

Comment:
interesting! The one with future saw a decent gain by avoiding vec!, but the other has negligible gains!

Reply:
I think the compiler might be optimizing away the creation of `vec` in the non-futures case, since it knows the `NoopExporter` doesn't do anything with it anyway. It probably cannot do that optimization for the one with Future.

@lalitb I think we would see the perf gain for the without future case as well, if you `black_box` the argument to export method.


```diff
fn emit(&self, record: &mut LogRecord, library: &InstrumentationLibrary) {
        let mut exporter = self.exporter.lock().expect("lock error");
-       futures_executor::block_on(exporter.export(vec![(record, library)]));
+       futures_executor::block_on(exporter.export(black_box(vec![(record, library)])));
    }
    
fn emit(&self, record: &mut LogRecord, library: &InstrumentationLibrary) {
        self.exporter
            .lock()
            .expect("lock error")
-           .export(vec![(record, library)]));
+           .export(black_box(vec![(record, library)]));
    }

Reply:
@lalitb would you like to update benchmarks? If not, we can merge now, and tweak benchmark in future.

Reply:
Yes will update it later, lets merge it now.

Reply:
I don't see any difference after adding the black_box. Probably, the compiler is not optimizing it out.

Reply:
Did you add the `black_box` to the main branch or this PR? It won't help with this PR as you have already removed the `vec!` creation. If you add `black_box` to the main branch, then you should see a higher baseline for the non-future case and then we would see the perf improvement from this PR.

Reply:
Thanks @utpilla , yes adding black_box to `main` gives result:
LogExporterWithFuture: 121
LogExporterWithoutFuture: 100

And with PR (LogBatch changes):
LogExporterWithFuture: 112
LogExporterWithoutFuture: 92